src/l3.c

```diff
@@ -112,6 +133,7 @@ l3__log_simple(uint32_t loc, const char *msg, const uint64_t arg1, const uint64_
 {
     int idx = __sync_fetch_and_add(&l3_log->idx, 1) % L3_MAX_SLOTS;
     l3_log->slots[idx].tid = l3_mytid();
+    l3_log->slots[idx].loc = 0;
```

Comment:
By default, `loc` will be 0.

The user-passed-in argument `loc` value will be stashed away in this field on L138 below.

Reply:
Did you measure whether there's a performance impact here? Maybe it will be zero overhead, but it might add a few ns.

Reply:
(It's ok if it does add a few ns btw, it's just that we should have a handle on where the time is doing.)

I profiled `printf()` and if you're redicting to a file it's actually quicker than I was expecting -- a simple `printf()` took about 40s and with a couple of integers in the format it took about 140ns. So L3 is still a lot quicker, but not actually by as much as I was expecting.

Reply:
We do need this `loc` field to be initialized to something when LOC is enabled. 

I went with your prior comment that `mmap()` will zero-out memory, so have changed the code to check that `assert(loc ==0)` under `DEBUG` build mode.

The new cod chunk will look like this:

```
138 #ifdef L3_LOC_ENABLED
139     l3_log->slots[idx].loc = (loc_t) loc;
140 #else
141 #ifdef DEBUG
142     assert(l3_log->slots[idx].loc == 0);
143 #endif  // DEBUG
144 #endif
```